cpp/arcticdb/storage/mock/s3_mock_client.cpp

```diff
@@ -109,32 +113,42 @@ S3Result<std::monostate> MockS3Client::put_object(
         return {*maybe_error};
     }
 
-    if (header == PutHeader::IF_NONE_MATCH && s3_contents.contains({bucket_name, s3_object_name})) {
+    auto pos = s3_contents_.find({bucket_name, s3_object_name});
+    if (header == PutHeader::IF_NONE_MATCH && pos != s3_contents_.end() && pos->second.has_value()) {
         return {precondition_failed_error};
     }
 
-    s3_contents.insert_or_assign({bucket_name, s3_object_name}, std::move(segment));
+    // When we write Segments we occasionally don't fill in the size_. This is fine as it's not needed for writing.
+    // However, it is required when reading so we need to calculate it. It's easier to do on write.
+    auto size[[maybe_unused]] = segment.calculate_size();
```

Comment:
the attribute is in a weird place, I would expected it at the start of the line?